pkg/internal/apis/certmanager/validation/certificate_test.go

```diff
@@ -563,6 +564,58 @@ func TestValidateCertificate(t *testing.T) {
 				field.Invalid(fldPath.Child("revisionHistoryLimit"), int32(0), "must not be less than 1"),
 			},
 		},
+		"maxPathLen positive and isCA set to true": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "testcn",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+					MaxPathLen: pointer.Int32(2),
+					IsCA:       true,
+				},
+			},
+			a:    someAdmissionRequest,
+			errs: []*field.Error{},
+		},
+		"maxPathLen zero and isCA set to true": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "testcn",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+					MaxPathLen: pointer.Int32(0),
+					IsCA:       true,
+				},
+			},
+			a:    someAdmissionRequest,
+			errs: []*field.Error{},
+		},
+		"maxPathLen set but not isCA": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "testcn",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+					MaxPathLen: pointer.Int32(2),
+					IsCA:       false,
+				},
+			},
+			a:    someAdmissionRequest,
+			errs: []*field.Error{field.Required(fldPath.Child("isCA"), "must set isCA to true when maxPathLen is set")},
+		},
+		"negative maxPathLen should be an error": {
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					CommonName: "testcn",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+					MaxPathLen: pointer.Int32(-2),
+					IsCA:       true,
+				},
+			},
+			a:    someAdmissionRequest,
+			errs: []*field.Error{field.Invalid(fldPath.Child("maxPathLen"), int32(-2), "maxPathLen must be either null, zero or a positive integer")},
```

Comment:
~suggestion: Add a validation test for nil MaxPathLen~

(Actually on reflection, I assume this is covered by other tests)

Reply:
> (Actually on reflection, I assume this is covered by other tests)

Yeah I think we have this covered - every test which doesn't explicitly set MaxPathLen is testing this; I suspect it'd be redundant to add another.